src/test/java/com/exasol/adapter/dialects/ScalarFunctionsAbstractIT.java

```diff
@@ -0,0 +1,495 @@
+     * define explicit parameters here. You can define multiple parameter combinations here (by adding multiple list
+     * entries). By that you can test a function with different values.
+     */
+    private static final Map<ScalarFunctionCapability, List<String>> EXPLICIT_PARAMETERS = Map.of(//
+            TO_YMINTERVAL, List.of("'3-11'"), //
+            NUMTODSINTERVAL, List.of("3.2, 'HOUR'"), //
+            NUMTOYMINTERVAL, List.of("3.5, 'YEAR'"), //
+            CONVERT_TZ, List.of("TIMESTAMP '2012-05-10 12:00:00', 'UTC', 'Europe/Berlin'"), //
+            JSON_VALUE, List.of("'{\"name\" : \"Smith\"}', '$.name'"), //
+            EXTRACT,
+            List.of("SECOND FROM TIMESTAMP '2000-10-01 12:22:59.123'", "MONTH FROM TIMESTAMP '2000-10-01 12:22:59.123'",
+                    "DAY FROM TIMESTAMP '2000-10-01 12:22:59.123'"), //
+            DATE_TRUNC, List.of("'month', DATE '2006-12-31'", "'day', DATE '2006-12-31'"), //
+            POSIX_TIME, List.of("'1970-01-01 00:00:01'"), //
+            ST_TRANSFORM, List.of("CAST('POINT(1 2)' AS GEOMETRY (4326)), 2163"), //
+            CAST, List.of("'1' as INTEGER")// TODO add a better test here
+    );
+
+    /**
+     * Some functions have different names than their capabilities. For that we define a mapping here. If a function is
+     * not listed, we assume, it has the same name as it's capability.
+     */
+    private static final Map<ScalarFunctionCapability, String> RENAME_FUNCTIONS = Map.of(MIN_SCALE, "MIN", //
+            NEG, "!"//
+    );
+
+    private static final int MAX_NUM_PARAMETERS = 4;
+    private static final int BATCH_SIZE = 500;
+    private static final String LOCAL_COPY_TABLE_NAME = "EXASOL.LOCAL_COPY";
+    private static final Logger LOGGER = LoggerFactory.getLogger(ScalarFunctionsAbstractIT.class);
+
+    /**
+     * One list for each level of parameters, that contains a list of parameter permutations. Each parameter permutation
+     * is a string with comma separated parameters.
+     */
+    private static List<List<String>> parameterCombinations;
+    private String fullyQualifiedNameOfArbitraryVsTable;
+    private ScalarFunctionsParameterCache parameterCache;
+
+    static Stream<Arguments> getScalarFunctions() {
+        return Arrays.stream(ScalarFunctionCapability.values())//
+                .filter(function -> !EXCLUDES.contains(function) && !FUNCTIONS_WITH_NO_PARENTHESIS.contains(function)
+                        && !function.name().startsWith("ST_"))//
+                .map(Arguments::of);
+    }
+
+    /**
+     * Permute the literals.
+     *
+     * @return permutations
+     */
+    private static List<List<String>> generateParameterCombinations(final List<String> atoms) {
+        final List<List<String>> combinations = new ArrayList<>(MAX_NUM_PARAMETERS + 1);
+        combinations.add(List.of(""));
+        for (int numParameters = 1; numParameters <= MAX_NUM_PARAMETERS; numParameters++) {
```

Comment:
otherwise, it's hard to name the variable. It's the current count of parameters. `previousIterationParameters` is the parameters of the run before, so I subtract one.